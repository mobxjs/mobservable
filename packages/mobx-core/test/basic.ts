import {cell, autorun, MobxState} from "../index"
import * as mobxCore from "../index";

test("autorun should work", () => {
    const context = new MobxState();
    const a = cell(context, 3);
    let v;

    expect(a.get()).toBe(3);
    const d = autorun(context, () => { v = a.get() });

    a.set(5)
    expect(v).toBe(5);
    d();
})

test("expose correct api", () => {
    const api = Object.keys(mobxCore).sort();
    expect(api).toMatchSnapshot();
})
