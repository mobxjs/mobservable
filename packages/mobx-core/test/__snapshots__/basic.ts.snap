// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expose correct api 1`] = `
Array [
  "Atom",
  "EMPTY_ARRAY",
  "EMPTY_OBJECT",
  "MobxState",
  "Reaction",
  "areBothNaN",
  "autorun",
  "cell",
  "computed",
  "createAction",
  "createInstanceofPredicate",
  "defaultComparer",
  "fail",
  "getDependencyTree",
  "getObserverTree",
  "hasOwnProperty",
  "identity",
  "identityComparer",
  "invariant",
  "isAtom",
  "isComputedValue",
  "isComputingDerivation",
  "isObject",
  "isObservableValue",
  "isPlainObject",
  "joinStrings",
  "noop",
  "onReactionError",
  "once",
  "primitiveSymbol",
  "reaction",
  "runInAction",
  "toPrimitive",
  "transaction",
  "unique",
  "untracked",
  "useStrict",
  "when",
]
`;
